.github/workflows/e2e-tests.yml

```diff
@@ -184,6 +184,36 @@ jobs:
           - script: |
               cd e2e/env-int-key
               env 1=1 ../../bin/phpstan analyse test.php
+          - script: |
+              cd e2e/trait-caching
+              ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
+              ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
+          - script: |
+              cd e2e/trait-caching
+              ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
+              patch -b data/TraitOne.php < TraitOne.patch
+              OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
+              echo "$OUTPUT"
+              [ $(echo "$OUTPUT" | wc -l) -eq 1 ]
+              grep 'Method TraitsCachingIssue\\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
```

Comment:
I am not sure its worth it. 

Would the goal be running the tests in a single job?

We would need to start clearing temp folder etc, so it will get more complicated.

Also not sure the assertions would be more readable.

Could you give me a few more directions what you would expect?

Reply:
I will start a new PR to give bashunit a try.